Show default node list after clearing search input

AustinMroz opened the pull request:
![searchbox-clear](https://github.com/user-attachments/assets/94737766-bf96-4f03-86a0-70057b3b0e81)
Resolves #4887 

<details>
<summary>The dirty details</summary>
There's really not a cleaner workaround here. Prime vue is hardcoded to hide results when a query is received with length 0. With our search box, we never want completions not to be shown and the sanest, if not the only viable solution, is to simply block the hiding completely.

Future TODO:
- Completely remove the reFocusInput jank. If we can make the search box a frame more responsive, we should.

```ts
onInput(event) {
    if (this.typeahead) {
        if (this.searchTimeout) {
            clearTimeout(this.searchTimeout);
        }

        let query = event.target.value;

        if (!this.multiple) {
            this.updateModel(event, query);
        }

        if (query.length === 0) {
            this.hide();
            this.$emit('clear');
        } else {
            if (query.length >= this.minLength) {
                this.focusedOptionIndex = -1;

                this.searchTimeout = setTimeout(() => {
                    this.search(event, query, 'input');
                }, this.delay);
            } else {
                this.hide();
            }
        }
    }
}
hide(isFocus) {
    const _hide = () => {
        this.$emit('before-hide');
        this.dirty = isFocus;
        this.overlayVisible = false;
        this.clicked = false;
        this.focusedOptionIndex = -1;

        isFocus && focus(this.multiple ? this.$refs.focusInput : this.$refs.focusInput?.$el);
    };

    setTimeout(() => {
        _hide();
    }, 0); // For ScreenReaders
}
```
</details>

┆Issue is synchronized with this [Notion page](https://www.notion.so/PR-6231-Show-default-node-list-after-clearing-search-input-2956d73d36508134960df537c7409646) by [Unito](https://www.unito.io)

AustinMroz committed:
Fix empty searchbox results

DrJKL commented on src/components/searchbox/NodeSearchBox.vue:
```diff
@@ -140,7 +142,11 @@ const reFocusInput = async () => {
 
-onMounted(reFocusInput)
+onMounted(() => {
+  acpref.value.hide = () => {}
```
Why not set this in the template?

AustinMroz committed:
Fix initial focus, naming, strange deselection bug

Under some circumstances, seemingly related to making a selection when
only one option exists, clicking the options would clear the searchbox
instead of adding the node to the graph.

AustinMroz commented on src/components/searchbox/NodeSearchBox.vue:
```diff
@@ -140,7 +141,13 @@ const reFocusInput = async () => {
+  inputElement ??= document.getElementById(inputId) as HTMLInputElement
+  if (inputElement) inputElement.focus()
+  autoCompletePlus.value.hide = () => search('')
+  search('')
```
A fairly lengthy journey of frustration. I started with an `@clear="search('')` and was excited to see that it would cause the computed list of options to update. I came real close to opening that as a one line fix only to see that, of course, the problem wasn't that easy. 

From there, I started doing debugging on the actual surrounding code. The relevant fragments are included in the top post, but one of `hide` or `onInput` needed to be changed and hide was simpler.

Swapping `hide` to call `search`('') instead of a no-op was a later commit to solve edge cases with options showing, but not being added to the graph on click.
